zulipterminal/ui_tools/messages.py

```diff
@@ -91,16 +92,24 @@ def __init__(self, message: Message, model: "Model", last_message: Any) -> None:
             if self._is_private_message_to_self():
                 recipient = self.message["display_recipient"][0]
                 self.recipients_names = recipient["full_name"]
+                self.recipients_names += (
+                    " (guest)"
+                    if self.model.show_guest_suffix(user_id=recipient["id"])
+                    else ""
+                )
                 self.recipient_emails = [self.model.user_email]
                 self.recipient_ids = [self.model.user_id]
             else:
-                self.recipients_names = ", ".join(
-                    [
-                        recipient["full_name"]
-                        for recipient in self.message["display_recipient"]
-                        if recipient["email"] != self.model.user_email
-                    ]
-                )
+                self.recipients_names = ""
+                recipients = []
+                for recipient in self.message["display_recipient"]:
+                    if recipient["email"] != self.model.user_email:
+                        recipient_name = self.recipients_names = recipient["full_name"]
+                        if self.model.show_guest_suffix(user_id=recipient["id"]):
+                            recipient_name += " (guest)"
+                        recipients.append(recipient_name)
+                self.recipients_names = ", ".join(recipients)
```

Comment:
This works fine but you can also maintain the original python comprehension and only add the suffix to the `recipient_name` with a conditional, similar to how it's done above in the private message conditional block